dp_check/dp_check.go

```diff
@@ -142,6 +227,35 @@ func main() {
 	if syscall.Getuid() != 0 {
 		infoLog.Println("Not running as root, some checks may fail.")
 	}
+
+	// Check if dp_check is running on a VM
+	runCheck("Running on GCE VM", func() error {
+		if !isRunningOnGCP() {
+			return fmt.Errorf(`dp_check is not running on a GCE VM`)
```

Comment:
nit: to be pedantic, can we add "this tool will not work as intended"

Reply:
Done